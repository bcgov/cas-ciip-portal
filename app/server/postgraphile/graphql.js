const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
