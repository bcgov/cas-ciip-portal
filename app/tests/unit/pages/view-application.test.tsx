import React from 'react';
import {shallow} from 'enzyme';
import ViewApplication from 'pages/reporter/view-application';
import {viewApplicationQueryResponse} from 'viewApplicationQuery.graphql';

const query: viewApplicationQueryResponse['query'] = {
  ' $fragmentRefs': {
    ApplicationDetailsContainer_query: true,
    ReviseApplicationButtonContainer_query: true
  },
  session: {
    ' $fragmentRefs': {
      defaultLayout_session: true
    }
  },
  application: {
    applicationRevisionStatus: {
      applicationRevisionStatus: 'SUBMITTED'
    },
    ' $fragmentRefs': {
      ReviseApplicationButtonContainer_application: true,
      ApplicationDetailsContainer_application: true
    },
    orderedFormResults: {
      edges: [
        {
          node: {
            id: 'abc',
            ' $fragmentRefs': {
              ApplicationCommentsContainer_formResult: true
            }
          }
        }
      ]
    }
  }
};

// It matches the last accepted Snapshot
it('It matches the last accepted Snapshot', () => {
  const wrapper = shallow(<ViewApplication query={query} router={null} />);
  expect(wrapper).toMatchSnapshot();
});

it('It passes a query to the ApplicationDetailsComponent component', () => {
  const wrapper = shallow(<ViewApplication query={query} router={null} />);
  expect(
    wrapper.find('Relay(ApplicationDetailsComponent)').first().prop('query')
  ).toBe(query);
});
