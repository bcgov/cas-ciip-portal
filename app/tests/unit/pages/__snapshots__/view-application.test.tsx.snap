// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches the last accepted Snapshot 1`] = `
<Relay(DefaultLayout)
  allowedGroups={
    Array [
      "User",
    ]
  }
  session={
    Object {
      " $fragmentRefs": Object {
        "defaultLayout_session": true,
      },
    }
  }
  showSubheader={true}
  title="Summary of your application"
>
  <ForwardRef
    noGutters={false}
  >
    <Col
      md={8}
    >
      <Relay(ApplicationDetailsComponent)
        application={
          Object {
            " $fragmentRefs": Object {
              "ApplicationDetailsContainer_application": true,
              "ReviseApplicationButtonContainer_application": true,
            },
            "applicationRevisionStatus": Object {
              "applicationRevisionStatus": "SUBMITTED",
            },
            "orderedFormResults": Object {
              "edges": Array [
                Object {
                  "node": Object {
                    " $fragmentRefs": Object {
                      "ApplicationCommentsContainer_formResult": true,
                    },
                    "id": "abc",
                  },
                },
              ],
            },
          }
        }
        query={
          Object {
            " $fragmentRefs": Object {
              "ApplicationDetailsContainer_query": true,
            },
            "application": Object {
              " $fragmentRefs": Object {
                "ApplicationDetailsContainer_application": true,
                "ReviseApplicationButtonContainer_application": true,
              },
              "applicationRevisionStatus": Object {
                "applicationRevisionStatus": "SUBMITTED",
              },
              "orderedFormResults": Object {
                "edges": Array [
                  Object {
                    "node": Object {
                      " $fragmentRefs": Object {
                        "ApplicationCommentsContainer_formResult": true,
                      },
                      "id": "abc",
                    },
                  },
                ],
              },
            },
            "session": Object {
              " $fragmentRefs": Object {
                "defaultLayout_session": true,
              },
            },
          }
        }
        review={false}
      />
    </Col>
    <Col
      md={4}
    >
      <Relay(ApplicationCommentsComponent)
        formResult={
          Object {
            " $fragmentRefs": Object {
              "ApplicationCommentsContainer_formResult": true,
            },
            "id": "abc",
          }
        }
        key="abc"
        review={false}
      />
    </Col>
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  />
</Relay(DefaultLayout)>
`;
